packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -362,24 +350,13 @@ function buildBodyParameter(
   }
 
   if (bodyParameter.type.type === "model") {
-    const bodyParts: string[] = [];
-    for (const param of getAllProperties(bodyParameter?.type).filter(
-      (p) => !p.readonly
-    ) ?? []) {
-      if (param.type.type === "model" && isRequired(param)) {
-        bodyParts.push(
-          `"${param.restApiName}": {${getRequestModelMapping(
-            param.type,
-            param.clientName,
-            importSet
-          ).join(",\n")}}`
-        );
-      } else {
-        bodyParts.push(getParameterMap(param, importSet));
-      }
-    }
+    const bodyParts: string[] = getRequestModelMapping(
+      bodyParameter.type,
+      bodyParameter.clientName,
+      importSet
+    );
```

Comment:
As the model is unflattened, we can just getRequestModelMapping without considering if any of those properties are coming from required signature or option bag.